filesystem/fs.go

```diff
@@ -76,11 +76,13 @@ func (cfs CFS) create(parentID uint64, name string, node *Node) error {
 	if err != nil {
 		return err
 	}
-	const sql = `
-INSERT INTO fs.inode VALUES ($1, $2);
-INSERT INTO fs.namespace VALUES ($3, $4, $1);
-`
-	if _, err := tx.Exec(sql, node.ID, inode, parentID, name); err != nil {
+	const insertNode = `INSERT INTO fs.inode VALUES ($1, $2)`
+	if _, err := tx.Exec(insertNode, node.ID, inode); err != nil {
+		_ = tx.Rollback()
+		return err
+	}
+	const insertNamespace = `INSERT INTO fs.namespace VALUES ($1, $2, $3)`
+	if _, err := tx.Exec(insertNamespace, parentID, name, node.ID); err != nil {
```

Comment:
Hmm. The docs are subtle but I think you're right. They say

> The possible responses to Execute are the same as those described above for queries issued via simple query protocol, except that Execute doesn't cause ReadyForQuery or RowDescription to be issued.

I took the first half of this to mean that multiple responses are possible, but since a `RowDescription` is normally the delimiter between different result sets, I think this precludes returning multiple responses to an `Execute`.

I'd be fine with using CTEs (if we supported them) to perform the two inserts in parallel; it's the hierarchical relationship implied by the `RETURNING` clause that feels wrong to me. (But then the `RETURNING` clause also feels unfamiliar to me in general because I'm coming from a MySQL background).